WordPress/UITestsFoundation/Screens/ReaderScreen.swift

```diff
@@ -194,23 +210,54 @@ public class ReaderScreen: ScreenObject {
     }
 
     @discardableResult
-    public func verifySavedPosts(state: String, postLabel: String? = nil) -> Self {
+    public func verifySavedPosts(state: String, postLabel: String? = nil, file: StaticString = #file, line: UInt = #line) -> Self {
         if readerTable.cells.count > 0 {
-            XCTAssertEqual(readerTable.cells.firstMatch.label, postLabel, "Post displayed does not match saved post!")
-            XCTAssertEqual(readerTable.cells.count, 1, "There should only be 1 post!")
-            XCTAssertEqual(state, .withSavedPosts)
+            XCTAssertTrue(readerTable.cells.firstMatch.waitForExistence(timeout: 3), file: file, line: line)
+            XCTAssertEqual(readerTable.cells.firstMatch.label, postLabel, .postNotEqualSavedPostError, file: file, line: line)
+            XCTAssertEqual(readerTable.cells.count, 1, .postNotEqualOneError, file: file, line: line)
+            XCTAssertEqual(state, .withSavedPosts, file: file, line: line)
         } else {
-            XCTAssertTrue(noResultsView.waitForExistence(timeout: 3))
-            XCTAssertTrue(readerTable.label == .emptyListLabel)
-            XCTAssertEqual(state, .withoutSavedPosts)
+            XCTAssertTrue(noResultsView.waitForExistence(timeout: 3), file: file, line: line)
+            XCTAssertTrue(readerTable.label == .emptyListLabel, file: file, line: line)
+            XCTAssertEqual(state, .withoutSavedPosts, file: file, line: line)
         }
 
         return self
     }
+
+    public func likeFirstPost() -> Self {
+        likeButton.firstMatch.waitForIsHittable()
+        XCTAssertTrue(likeButton.firstMatch.label.hasPrefix(.postNotLiked))
+        likeButton.firstMatch.tap()
+
+        return self
+    }
+
+    public func verifyPostLikedOnFollowing(file: StaticString = #file, line: UInt = #line) -> Self {
+        XCTAssertTrue(readerTable.cells.firstMatch.waitForExistence(timeout: 3), file: file, line: line)
+        XCTAssertGreaterThan(readerTable.cells.count, 1, .postNotGreaterThanOneError, file: file, line: line)
```

Comment:
The fact the posts amount on "Following" tab should be greater than 1, and that on "Likes" tab it should be 1, looks like hardcoded test data (from mocks) into functions. Is this check (in both functions) really relevant for the likes test, taking into account that "Likes" tab will show one liked post no matter if it was liked or not (as long as we use mocks)?

Reply:
this is a good point! yeah because of mocks it will be 1, let me see if i can update the mocks so that it doesn't show the post in the likes tab before liking it.